webapp/xsl/v1-to-v2.xsl

```diff
@@ -0,0 +1,213 @@
+        encoding="UTF-8"
+        omit-xml-declaration="no"
+        indent="no"/>
+    
+    <xsl:template match="article">
+        <xsl:text disable-output-escaping='yes'>&lt;!DOCTYPE article PUBLIC "-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with MathML3 v1.2 20190208//EN"  "JATS-archivearticle1-mathml3.dtd"&gt;</xsl:text>
+        <xsl:choose>
+            <!-- do not do anything for articles which aren't insights -->
+            <xsl:when test="not(@article-type='article-commentary')">
+                <xsl:copy-of select="."/>
+            </xsl:when>
+            <!-- do not transform insight xml already in new format -->
+            <xsl:when test="descendant::article-meta//custom-meta[meta-name='elife-xml-version' and meta-value='2.0']">
+                <xsl:copy-of select="."/>
+            </xsl:when>
+            <!-- convert to 2.0 format -->
+            <xsl:otherwise>
+                <xsl:copy>
+                    <xsl:attribute name="article-type">
+                        <xsl:value-of select="./@article-type"/>
+                    </xsl:attribute>
+                    <xsl:attribute name="dtd-version">1.2</xsl:attribute>
+                    <xsl:attribute name="specific-use">version-of-record</xsl:attribute>
+                    <xsl:apply-templates/>
+                </xsl:copy>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+    
+    <xsl:template match="*|@*|text()|comment()|processing-instruction()">
+        <xsl:copy>
+            <xsl:apply-templates select="*|@*|text()|comment()|processing-instruction()"/>
+        </xsl:copy>
+    </xsl:template>
+    
+    <xsl:template match="abstract">
+        <xsl:for-each
+            select="./parent::article-meta/custom-meta-group/custom-meta[meta-name = 'Author impact statement']/meta-value">
+            <xsl:element name="abstract">
+                <xsl:attribute name="abstract-type">
+                    <xsl:value-of select="'toc'"/>
+                </xsl:attribute>
+                <xsl:element name="p">
+                    <xsl:apply-templates/>
+                </xsl:element>
+            </xsl:element>
+        </xsl:for-each>
+        <xsl:copy-of select="."/>
+    </xsl:template>
+    
+    <xsl:template match="article-categories">
+        <xsl:element name="article-categories">
+            <xsl:for-each select="./subj-group[@subj-group-type='display-channel']">
+                <xsl:element name="subj-group">
+                    <xsl:attribute name="subj-group-type">heading</xsl:attribute>
+                    <xsl:apply-templates select="./*"/>
+                </xsl:element>
+            </xsl:for-each>
+            <xsl:if test="./subj-group[@subj-group-type='heading']">
+                <xsl:element name="subj-group">
+                    <xsl:attribute name="subj-group-type">subject</xsl:attribute>
+                    <xsl:for-each select="./subj-group[@subj-group-type='heading']/subject">
```

Comment:
Sorry looks like this is an older version (needs to be `major-subject` instead of `subject`). 

I've updated the versions [here](https://app.zenhub.com/workspaces/libero-editor-5d56a6cfaa13c900012ad75b/issues/libero/editor/770), so please can you update both xsl files with those in that ticket?

Reply:
I've added the replaced download file, `<xsl:attribute name="subj-group-type">major-subject</xsl:attribute>`